Content.Server/Discord/DiscordLink.cs

```diff
@@ -0,0 +1,148 @@
+﻿using System.Diagnostics;
+using System.Threading.Tasks;
+using Content.Shared.CCVar;
+using Discord;
+using Discord.WebSocket;
+using Robust.Shared.Configuration;
+using LogMessage = Discord.LogMessage;
+
+namespace Content.Server.Discord;
+
+public sealed class DiscordLink : IPostInjectInit
+{
+    [Dependency] private readonly ILogManager _logManager = default!;
+    [Dependency] private readonly IConfigurationManager _configuration = default!;
+
+    public DiscordSocketClient? Client;
+    private ISawmill _sawmill = default!;
+
+    private string _guildId = string.Empty;
+
+    public void PostInject()
+    {
+        _sawmill = _logManager.GetSawmill("discord.link");
+
+        Client = new DiscordSocketClient(new DiscordSocketConfig()
```

Comment:
This should absolutely not be constructed from `PostInject()`. This should be from some initialize call, and only if a token is specified.